### ISSUE TITLE
Add EDV provider

### ISSUE THREAD

**Issue by DRK3** [hyperledger/aries-framework-go#2199]
REST API and EDV models must be decoupled so as to allow the EDV models to be used in other storage providers.

- [x] FormatProvider (done in #2210)
- [x] #2249 
- [x] #2273
- [x] #2208

**Comment by llorllale** [comment]
@DRK3 

**High-level Design**

`storage.Formatter`: formats documents to and from storages in a way that preserves the user's privacy vs. external storage providers

```go
package storage

type Formatter interface {
   FormatPair(k string, v []byte) ([]byte, error)
   ParseValue([]byte) ([]byte, error)
}
```

`edv.Formatter`: implements `storage.Formatter`

`edv.Store`: implements `storage.Store`

We will not have to add any new methods to `storage.Store`. We can delete `storage.Store.Query()`.

**Comment by DRK3** [comment]
Support for the https://github.com/trustbloc/edv ReturnFullDocumentsOnQuery and Batch extensions have also been implemented in #2355/#2356 and #2365.
